python/lsst/daf/butler/formatters/fileFormatter.py

```diff
@@ -178,6 +248,28 @@ def write(self, inMemoryDataset, fileDescriptor):
 
         return fileDescriptor.location.pathInStore
 
+    def toBytes(self, inMemoryDataset, fileDescriptor):
+        """Serialize the Dataset to bytes based on formatter.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            The Python object to serialize.
+        fileDescriptor : `FileDescriptor`
+            Identifies read type and parameters to be used for reading.
+
+        Returns
+        -------
+        serializedDataset : `bytes`
+            bytes representing the serialized dataset.
+        """
+        if not hasattr(self, '_toBytes'):
+            raise NotImplementedError("Type does not support reading from bytes.")
+
+        fileDescriptor.location.updateExtension(self.extension)
```

Comment:
Why does serializing to bytes result in the extension being updated on a file name that might not be used?

Reply:
Removed.